galsim/zernike.py

```diff
@@ -0,0 +1,398 @@
+        # Implement this by retrieving R[n, 0] coefficients of (r^2)^k and
+        # multiplying in the binomial (in r^2) expansion of ((r^2 - eps^2)/(1 - eps^2))^k
+        coefs = _zern_rho_coefs(n, 0)
+        for i, coef in enumerate(coefs):
+            if i % 2 == 1: continue
+            j = i // 2
+            more_coefs = (norm**j) * binomial(-eps**2, 1, j)
+            out[0:i+1:2] += coef*more_coefs
+    elif m == n:  # Equation (25)
+        norm = 1./np.sqrt(np.sum((eps**2)**np.arange(n+1)))
+        out[n] = norm
+    else:  # Equation (A1)
+        j = (n-m)//2
+        norm = np.sqrt((1-eps**2)/(2*(2*j+m+1) * _h(m,j,eps)))
+        out[m::2] = norm * _Q(m, j, eps)
+    return out
+_annular_zern_rho_coefs = LRU_Cache(__annular_zern_rho_coefs)
+
+
+class Zernike(object):
+    """A class to represent Zernike polynomial series
+    (http://en.wikipedia.org/wiki/Zernike_polynomials#Zernike_polynomials).
+
+    Zernike polynomials form an orthonormal basis over the unit circle.  The convention used here is
+    for the normality constant to equal the area of integration, which is pi for the unit circle.
+    I.e.,
+
+        \int_{unit circle} Z_i Z_j dA = \pi \delta_{i, j}.
+
+    Two generalizations of the unit circle Zernike polynomials are also available in this class:
+    annular Zernike polynomials, and polynomials defined over non-unit-radius circles.
+
+    Annular Zernikes are orthonormal over an annulus instead of a circle (see Mahajan, J. Opt. Soc.
+    Am. 71, 1, (1981)).  Similarly, the non-unit-radius polynomials are orthonormal over a region
+    with outer radius not equal to 1.  Taken together, these generalizations yield the
+    orthonormality condition:
+
+        \int_{annulus} Z_i Z_j dA = \pi (R_outer^2 - R_inner^2) \delta_{i, j}
+
+    where 0 < R_inner < R_outer indicate the inner and outer radii of the annulus over which the
+    polynomials are orthonormal.
+
+    The indexing convention for i and j is that from Noll, J. Opt. Soc. Am. 66, 207-2011(1976),
+    though offset by 1 because python is 0-indexed while the Noll indices start at 1.  For example,
+    the first few Zernike polynomials in terms of Cartesian coordinates x and y are
+
+        this index  |  Noll index  |  polynomial
+        -----------------------------------
+             0      |       1      |      1
+             1      |       2      |      2 x
+             2      |       3      |      2 y
+             3      |       4      | sqrt(3) (2 (x^2 + y^2) - 1)
+
+
+    @param coef   Zernike series coefficients.  Note that coef[i] corresponds to Z_(i+1) using the
+                  Noll index convention.  (I.e., coef[0] is 'piston', coef[3] is 'defocus', ...)
+    @param eps    Linear fractional obscuration for defining annular Zernike polynomials.
+                  [default: 0.0]
+    @param diam   Diameter over which polynomials are orthonormal.  [Default: 2.0 (for unit circle)]
```

Comment:
From the above docstring, I expected the parameters to be `R_inner` and `R_outer`, rather than `eps` and `diam`.  `eps` in particular was not intuitive to me without reading the param description.  

I'd suggest switching the parameters to those, which I thought were pretty clear in your description.